crypto/ed25519/ed25519.go

```diff
@@ -65,14 +64,14 @@ func (privKey PrivKeyEd25519) PubKey() crypto.PubKey {
 			break
 		}
 	}
-	if initialized {
-		var pubkeyBytes [PubKeyEd25519Size]byte
-		copy(pubkeyBytes[:], privKeyBytes[32:])
-		return PubKeyEd25519(pubkeyBytes)
+
+	if !initialized {
+		panic("pubkey bytes should always be initialized!")
```

Comment:
This may not be true with multi-language support for generating private keys. I'm in favor of keeping the old code here.

Reply:
So the only place we read in a priv key is the `priv_validator.json` and `node_key.json`. I think we can just make it a requirement that these come in the format that includes the pubkey? Then we don't have to worry here.

Think that's legit?

Reply:
Having the pubkey inside of the encoded private key for ed25519 is a design quirk they had to do due to the limited interface of cryptosystems. I think we should support using the normal format as well.